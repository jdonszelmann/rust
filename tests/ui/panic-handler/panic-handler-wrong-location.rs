//@ compile-flags:-C panic=abort

#![no_std]
#![no_main]

#[panic_handler] //~ ERROR `#[panic_handler]` is only valid on function
#[no_mangle]
static X: u32 = 42;

//~? ERROR `#[panic_handler]` required, but not found
