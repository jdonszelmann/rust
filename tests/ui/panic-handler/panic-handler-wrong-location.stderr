error: `#[panic_handler]` is only valid on functions
  --> $DIR/panic-handler-wrong-location.rs:6:1
   |
LL | #[panic_handler]
   | ^^^^^^^^^^^^^^^^

error: `#[panic_handler]` required, but not found
  --> $SRC_DIR/core/src/panic.rs:LL:COL
   |
   = note: expected because `#[panic_handler]` was declared here in crate `core`
   |
   = help: expected at least one implementation in crate `panic_handler_wrong_location` or any of its dependencies

error: aborting due to 2 previous errors

